cel/decls.go

```diff
@@ -1204,3 +1204,34 @@ func collectParamNames(paramNames map[string]struct{}, arg *Type) {
 		collectParamNames(paramNames, param)
 	}
 }
+
+func typeValueToKind(tv *types.TypeValue) (Kind, error) {
```

Comment:
What did you have in mind in terms of making `typeValueToKind` more general? I think all types with both a defined `Kind` and `TypeValue` singleton are already supported - there's `DynKind` but no equivalent `DynType` singleton for instance.

Reply:
I don't believe that `DynType`, `ObjectType`, or `OptionalType` and a few others are supported in this list because they're not supported as string formatting arguments, so it's worth either supporting all type to kind mappings here, or moving the function to a place where this mapping of type to kind is suitable for only those types which support string formatting.

Reply:
It didn't occur to me to use `TypeValue.TypeName()` to attempt to map `TypeValue` to `Kind` in cases where a singleton isn't available, though this still leaves `StructKind` and `TypeParamKind` without a mapping since as far as I can gather, both of those have caller-set values for their names. Is there some other solution for those two `Kind` cases?